src/main/java/org/isf/menu/rest/UserSettingController.java

```diff
@@ -0,0 +1,115 @@
+ */
+package org.isf.menu.rest;
+
+import javax.validation.Valid;
+
+import org.isf.menu.dto.UserSettingDTO;
+import org.isf.menu.manager.UserSettingManager;
+import org.isf.menu.mapper.UserSettingMapper;
+import org.isf.menu.model.UserSetting;
+import org.isf.shared.exceptions.OHAPIException;
+import org.isf.utils.exception.OHServiceException;
+import org.isf.utils.exception.model.OHExceptionMessage;
+import org.slf4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import io.swagger.annotations.Api;
+
+@RestController
+@Api(value = "/usersettings", produces = MediaType.APPLICATION_JSON_VALUE)
+public class UserSettingController {
+
+	private static final Logger LOGGER = org.slf4j.LoggerFactory.getLogger(UserSettingController.class);
+	
+	@Autowired
+	private UserSettingMapper userSettingMapper;
+	
+	@Autowired
+	private UserSettingManager userSettingManager;
+	
+	public UserSettingController(UserSettingMapper userSettingMapper, UserSettingManager userSettingManager) {
+		this.userSettingMapper = userSettingMapper;
+		this.userSettingManager = userSettingManager;
+	}
+	
+	/**
+	 * Create or update {@link UserSettingDTO}.
+	 * @param userSettingDTO - the {@link UserSettingDTO} to insert
+	 * @return {@link UserSettingDTO} if the userSetting has been inserted, null otherwise.
+	 * @throws OHServiceException 
+	 */
+	@PostMapping(value = "/usersettings", produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<UserSettingDTO> newUserSetting(@Valid @RequestBody UserSettingDTO userSettingDTO) throws OHServiceException {
+		LOGGER.info("Attempting to create or update a UserSetting");
+		String userName = SecurityContextHolder.getContext().getAuthentication().getName();
+		UserSetting userSetting = userSettingManager.getUserSettingDashboard(userName, userSettingDTO.getConfigName());
+		UserSetting isCreated = new UserSetting();
+		if (userSetting != null) {
+			LOGGER.info("update a UserSetting");
+			userSetting.setConfigValue(userSettingDTO.getConfigValue());
+			isCreated = userSettingManager.updateUserSetting(userSetting);
+		} else {
+			LOGGER.info("create a UserSetting");
+			userSetting = userSettingMapper.map2Model(userSettingDTO);
+			userSetting.setUser(userName);
+			isCreated = userSettingManager.newUserSetting(userSetting);
+		}
+		
+		if (isCreated == null) {
+			LOGGER.info("UserSetting is not created!");
+            throw new OHAPIException(new OHExceptionMessage("UserSetting not created."));
```

Comment:
the formatting from here to the end of the method looks wrong.

Reply:
please i don't know why we have this formatting here in my IDE on don't have it and I don't how I can solve it .

Reply:
If you follow https://github.com/informatici/openhospital-api#code-style you should be ready to go